### DISCUSSION TITLE
Should we remove the `deltaCount` check?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#96] [General]
Given that  the `createInternal` function calls `checkSegmentCounts` function that checks if the counts of segments match and if the counts are less than the `MAX_SEGMENT_COUNT`, I don't think this check from the `createWithDuration` function is still necessary:

https://github.com/sablierhq/v2-core/blob/5f13c560f6a3c99a10899ffc95907a8bc7414d84/src/SablierV2Pro.sol#L238-L241

**Comment by PaulRBerg** [comment]
This is an interesting idea - I have to sleep on it though.

You're right that the call would revert anyway. But I'm trying to think if there's any way this would open up an exploit .. Lemme think.

**Comment by PaulRBerg** [comment]
Created an issue for this, locking.

https://github.com/sablierhq/v2-core/issues/99
